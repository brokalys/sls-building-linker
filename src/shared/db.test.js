const { query } = require('serverless-mysql');
const db = require('./db');

jest.mock('serverless-mysql');

describe('updatePropertyBuildingId', () => {
  afterEach(jest.clearAllMocks);

  test('constructs the correct query', () => {
    db.updatePropertyBuildingId(111, 222);

    expect(query).toBeCalled();
    expect(query.mock.calls[0][0]).toMatchSnapshot();
  });
});

describe('createPropertyBuildingLink', () => {
  afterEach(jest.clearAllMocks);

  test('constructs the correct query', () => {
    db.createPropertyBuildingLink(111, 222, 'latlng');

    expect(query).toBeCalled();
    expect(query.mock.calls[0][0]).toMatchSnapshot();
  });
});

describe('findBuildingIdByLatLng', () => {
  afterEach(jest.clearAllMocks);

  test('constructs the correct query', () => {
    db.findBuildingIdByLatLng(56.123, 24.222);

    expect(query).toBeCalled();
    expect(query.mock.calls[0][0]).toMatchSnapshot();
  });

  test('returns nothing if nothing found', async () => {
    query.mockResolvedValueOnce([]);

    const output = await db.findBuildingIdByLatLng(56.123, 24.222);

    expect(output).toBeUndefined();
  });

  test('returns the row id if something is found', async () => {
    query.mockResolvedValueOnce([{ id: 333 }]);

    const output = await db.findBuildingIdByLatLng(56.123, 24.222);

    expect(output).toBe(333);
  });
});

describe('findVzdBuildingIdByLatLng', () => {
  afterEach(jest.clearAllMocks);

  test('constructs the correct query', () => {
    db.findVzdBuildingIdByLatLng(56.123, 24.222);

    expect(query).toBeCalled();
    expect(query.mock.calls[0][0]).toMatchSnapshot();
  });

  test('returns nothing if nothing found', async () => {
    query.mockResolvedValueOnce([]);

    const output = await db.findVzdBuildingIdByLatLng(56.123, 24.222);

    expect(output).toBeUndefined();
  });

  test('returns the row id if something is found', async () => {
    query.mockResolvedValueOnce([{ id: 333 }]);

    const output = await db.findVzdBuildingIdByLatLng(56.123, 24.222);

    expect(output).toBe(333);
  });
});

describe('findVzdBuildingIdByLocation', () => {
  afterEach(jest.clearAllMocks);

  test('constructs the correct query', () => {
    db.findVzdBuildingIdByLocation({
      city: 'riga',
      street: 'brivibas',
      housenumber: '12',
    });

    expect(query).toBeCalled();
    expect(query.mock.calls[0][0]).toMatchSnapshot();
  });

  test('returns nothing if nothing found', async () => {
    query.mockResolvedValueOnce([]);

    const output = await db.findVzdBuildingIdByLocation({
      city: 'riga',
      street: 'brivibas',
      housenumber: '12',
    });

    expect(output).toBeUndefined();
  });

  test('returns the row id if something is found', async () => {
    query.mockResolvedValueOnce([{ id: 333 }]);

    const output = await db.findVzdBuildingIdByLocation({
      city: 'riga',
      street: 'brivibas',
      housenumber: '12',
    });

    expect(output).toBe(333);
  });
});
