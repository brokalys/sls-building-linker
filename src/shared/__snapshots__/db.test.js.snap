// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createPropertyBuildingLink constructs the correct query 1`] = `
Object {
  "sql": "
      REPLACE INTO property_building_links
      SET ?
   ",
  "values": Object {
    "link_type": "latlng",
    "property_id": 111,
    "vzd_building_id": 222,
  },
}
`;

exports[`createPropertyLandLink constructs the correct query 1`] = `
Object {
  "sql": "
      REPLACE INTO property_land_links
      SET ?
   ",
  "values": Object {
    "link_type": "latlng",
    "property_id": 111,
    "vzd_land_id": 222,
  },
}
`;

exports[`findVzdBuildingIdByLatLng constructs the correct query 1`] = `
Object {
  "sql": "
      SELECT id, ST_DISTANCE(bounds, POINT(?, ?)) as distance
      FROM vzd_buildings
      WHERE MBRIntersects(bounds, LineString(Point(?, ?), Point(?, ?)))
      AND is_usable = 1
      ORDER BY distance ASC
      LIMIT 1
   ",
  "values": Array [
    56.123,
    24.222,
    56.122499999999995,
    24.221500000000002,
    56.1235,
    24.2225,
  ],
}
`;

exports[`findVzdBuildingIdByLocation constructs the correct query 1`] = `
Object {
  "sql": "
      SELECT id
      FROM vzd_buildings
      WHERE is_usable = 1
        AND city = ?
        AND street = ?
        AND house_number = ?
      ORDER BY id ASC
      LIMIT 1
   ",
  "values": Array [
    "riga",
    "brivibas",
    "12",
  ],
}
`;

exports[`findVzdIdByCadastreNumber using "building" category constructs the correct query 1`] = `
Object {
  "sql": "
      SELECT id
      FROM vzd_buildings
      WHERE is_usable = 1
        AND cadastral_designation = ?
      ORDER BY id ASC
      LIMIT 1
   ",
  "values": Array [
    "01000762043",
  ],
}
`;

exports[`findVzdIdByCadastreNumber using "land" category constructs the correct query 1`] = `
Object {
  "sql": "
      SELECT id
      FROM vzd_land
      WHERE is_usable = 1
        AND cadastral_designation = ?
      ORDER BY id ASC
      LIMIT 1
   ",
  "values": Array [
    "01000762043",
  ],
}
`;

exports[`findVzdLandIdByLatLng constructs the correct query 1`] = `
Object {
  "sql": "
      SELECT id, ST_DISTANCE(bounds, POINT(?, ?)) as distance
      FROM vzd_land
      WHERE MBRIntersects(bounds, LineString(Point(?, ?), Point(?, ?)))
      AND is_usable = 1
      ORDER BY distance ASC
      LIMIT 1
   ",
  "values": Array [
    56.123,
    24.222,
    56.122499999999995,
    24.221500000000002,
    56.1235,
    24.2225,
  ],
}
`;
